lua/ocaml/config.lua

```diff
@@ -0,0 +1,48 @@
+local vim = vim
+local M = {}
+
+---@class OCamlConfigKeymaps
+---@field jump_next_hole string|?
+---@field jump_prev_hole string|?
+---@field construct string|?
+---@field jump string|?
+---@field phrase_prev string|?
+---@field phrase_next string|?
+---@field infer string|?
+---@field switch_ml_mli string|?
+---@field type_enclosing string|?
+---@field type_enclosing_grow string|?
+---@field type_enclosing_shrink string|?
+---@field type_enclosing_increase string|?
+---@field type_enclosing_decrease string|?
+
+---@class ocaml.config.OCamlConfig
+---@field keymaps OCamlConfigKeymaps
+
+--- Default values
+--- @type ocaml.config.OCamlConfig
+local default = {
+  keymaps = {
+    jump_next_hole = "<leader>n",
+    jump_prev_hole = "<leader>p",
+    construct = "<leader>c",
+    jump = "<leader>j",
+    phrase_prev = "<leader>pp",
+    phrase_next = "<leader>pn",
+    infer = "<leader>i",
+    switch_ml_mli = "<leader>s",
+    type_enclosing = "<leader>t",
+    type_enclosing_grow = "<Up>",
+    type_enclosing_shrink = "<Down>",
+    type_enclosing_increase = "<Right>",
+    type_enclosing_decrease = "<Left>",
+  },
+}
+
+--- Extends the configuration with user config
+---@return ocaml.config.OCamlConfig
+function M.init_config(user_config)
+  return vim.tbl_deep_extend("force", default, user_config or {})
```

Comment:
This function is creating trouble as providing a `keymaps = {}` still takes the default values. I need to investigate but maybe it should iterate over the pairs. @Lucccyo any idea about that? 😄

Reply:
I was also confused about that, because I didn't find a proper way to set up a default nothing and to allow the user to define their own keymaps. That is why I stayed with the defaults for the moment :smile:.
However, since this is really better than the defaults, I will also investigate how to handle it.